.travis.yml

```diff
@@ -28,9 +36,26 @@ cache:
 notifications:
   email: false
 
+before_install:
+  # Download Python packages.
+  - pip install coverage pycodestyle
+  - pip install py4j==0.10.7
+  # Download Spark
+  - mkdir $TRAVIS_BUILD_DIR/spark
+  - curl -O http://mirrors.gigenet.com/apache/spark/spark-$TEST_SPARK_VERSION/spark-$TEST_SPARK_VERSION-bin-hadoop2.7.tgz
+  - tar zxfC spark-$TEST_SPARK_VERSION-bin-hadoop2.7.tgz $TRAVIS_BUILD_DIR/spark
+
 install:
   - build/sbt assembly -Drepourl=http://nexus-private.hortonworks.com/nexus/content/groups/public/ -Dhadoop.version=3.0.0.3.0.0.0-SNAPSHOT -Dhive.version=3.0.0.3.0.0.0-SNAPSHOT
+  # sbt test:assembly generates a jar containing test classes complied. This is required to run the Python tests.
+  - build/sbt test:assembly -Drepourl=http://nexus-private.hortonworks.com/nexus/content/groups/public/ -Dhadoop.version=3.0.0.3.0.0.0-SNAPSHOT -Dhive.version=3.0.0.3.0.0.0-SNAPSHOT
```

Comment:
Ur, it would be enough to add `test:` at line 49.

Reply:
Actually, nope .. since I changed the full class path in this scope:

```
fullClasspath in (Test, assembly) := {
  ...
```

So, sbt test:assembly creates a jar containing only test classes. It's a bit different with Spark's I believe .. 

If I don't do this (and include test classes with the main code in a single jar), it causes a kind of dependency problems which needs more looks and more changes ..

Reply:
In that case, `test:package` will be the solution.